// @opentelemetry/core/build/esm/platform/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _node = require('./node');

Object.defineProperty(exports, 'RandomIdGenerator', {
  enumerable: true,
  get: function get() {
    return _node.RandomIdGenerator;
  }
});
Object.defineProperty(exports, 'SDK_INFO', {
  enumerable: true,
  get: function get() {
    return _node.SDK_INFO;
  }
});
Object.defineProperty(exports, '_globalThis', {
  enumerable: true,
  get: function get() {
    return _node._globalThis;
  }
});
Object.defineProperty(exports, 'getEnv', {
  enumerable: true,
  get: function get() {
    return _node.getEnv;
  }
});
Object.defineProperty(exports, 'getEnvWithoutDefaults', {
  enumerable: true,
  get: function get() {
    return _node.getEnvWithoutDefaults;
  }
});
Object.defineProperty(exports, 'hexToBase64', {
  enumerable: true,
  get: function get() {
    return _node.hexToBase64;
  }
});
Object.defineProperty(exports, 'otperformance', {
  enumerable: true,
  get: function get() {
    return _node.otperformance;
  }
});
Object.defineProperty(exports, 'unrefTimer', {
  enumerable: true,
  get: function get() {
    return _node.unrefTimer;
  }
});
