// @opentelemetry/resources/build/esm/detectors/platform/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _node = require('./node');

Object.defineProperty(exports, 'hostDetector', {
  enumerable: true,
  get: function get() {
    return _node.hostDetector;
  }
});
Object.defineProperty(exports, 'hostDetectorSync', {
  enumerable: true,
  get: function get() {
    return _node.hostDetectorSync;
  }
});
Object.defineProperty(exports, 'osDetector', {
  enumerable: true,
  get: function get() {
    return _node.osDetector;
  }
});
Object.defineProperty(exports, 'osDetectorSync', {
  enumerable: true,
  get: function get() {
    return _node.osDetectorSync;
  }
});
Object.defineProperty(exports, 'processDetector', {
  enumerable: true,
  get: function get() {
    return _node.processDetector;
  }
});
Object.defineProperty(exports, 'processDetectorSync', {
  enumerable: true,
  get: function get() {
    return _node.processDetectorSync;
  }
});
Object.defineProperty(exports, 'serviceInstanceIdDetectorSync', {
  enumerable: true,
  get: function get() {
    return _node.serviceInstanceIdDetectorSync;
  }
});
