// @opentelemetry/resources/build/esm/detectors/platform/node/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _HostDetector = require('./HostDetector');

Object.defineProperty(exports, 'hostDetector', {
  enumerable: true,
  get: function get() {
    return _HostDetector.hostDetector;
  }
});

var _HostDetectorSync = require('./HostDetectorSync');

Object.defineProperty(exports, 'hostDetectorSync', {
  enumerable: true,
  get: function get() {
    return _HostDetectorSync.hostDetectorSync;
  }
});

var _OSDetector = require('./OSDetector');

Object.defineProperty(exports, 'osDetector', {
  enumerable: true,
  get: function get() {
    return _OSDetector.osDetector;
  }
});

var _OSDetectorSync = require('./OSDetectorSync');

Object.defineProperty(exports, 'osDetectorSync', {
  enumerable: true,
  get: function get() {
    return _OSDetectorSync.osDetectorSync;
  }
});

var _ProcessDetector = require('./ProcessDetector');

Object.defineProperty(exports, 'processDetector', {
  enumerable: true,
  get: function get() {
    return _ProcessDetector.processDetector;
  }
});

var _ProcessDetectorSync = require('./ProcessDetectorSync');

Object.defineProperty(exports, 'processDetectorSync', {
  enumerable: true,
  get: function get() {
    return _ProcessDetectorSync.processDetectorSync;
  }
});

var _ServiceInstanceIdDetectorSync = require('./ServiceInstanceIdDetectorSync');

Object.defineProperty(exports, 'serviceInstanceIdDetectorSync', {
  enumerable: true,
  get: function get() {
    return _ServiceInstanceIdDetectorSync.serviceInstanceIdDetectorSync;
  }
});
