// @opentelemetry/core/build/esnext/platform/browser/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _environment = require('./environment');

Object.defineProperty(exports, 'getEnvWithoutDefaults', {
  enumerable: true,
  get: function get() {
    return _environment.getEnvWithoutDefaults;
  }
});
Object.defineProperty(exports, 'getEnv', {
  enumerable: true,
  get: function get() {
    return _environment.getEnv;
  }
});

var _globalThis2 = require('./globalThis');

Object.defineProperty(exports, '_globalThis', {
  enumerable: true,
  get: function get() {
    return _globalThis2._globalThis;
  }
});

var _hexToBase = require('./hex-to-base64');

Object.defineProperty(exports, 'hexToBase64', {
  enumerable: true,
  get: function get() {
    return _hexToBase.hexToBase64;
  }
});

var _RandomIdGenerator = require('./RandomIdGenerator');

Object.defineProperty(exports, 'RandomIdGenerator', {
  enumerable: true,
  get: function get() {
    return _RandomIdGenerator.RandomIdGenerator;
  }
});

var _performance = require('./performance');

Object.defineProperty(exports, 'otperformance', {
  enumerable: true,
  get: function get() {
    return _performance.otperformance;
  }
});

var _sdkInfo = require('./sdk-info');

Object.defineProperty(exports, 'SDK_INFO', {
  enumerable: true,
  get: function get() {
    return _sdkInfo.SDK_INFO;
  }
});

var _timerUtil = require('./timer-util');

Object.defineProperty(exports, 'unrefTimer', {
  enumerable: true,
  get: function get() {
    return _timerUtil.unrefTimer;
  }
});
